src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/ObjectHeader.cs

```diff
@@ -61,30 +83,37 @@ public static unsafe int GetHashCode(object o)
             fixed (MethodTable** ppMethodTable = &o.GetMethodTableRef())
```

Comment:
We should keep `GetHashCode` as it was unless the changes make it faster (seems unlikely). 
It would be ok if `TryGetHashCode` has some duplicated code in this case.

Reply:
I reverted the change to `GetHashCode` in favor of duplicating a small amount of code.